test/python/test_projectq_cpp_simulator.py

```diff
@@ -0,0 +1,184 @@
+# -*- coding: utf-8 -*-
+# pylint: disable=invalid-name,missing-docstring
+
+# Copyright 2017 IBM RESEARCH. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# =============================================================================
+
+import unittest
+import os
+import logging
+import random
+import numpy
+from scipy.stats import chi2_contingency
+import qiskit
+import qiskit.backends._projectq_simulator as projectq_simulator
+import qiskit.backends._qasmsimulator as qasm_simulator
+from qiskit import QuantumProgram
+from qiskit import QuantumRegister
+from qiskit import ClassicalRegister
+from qiskit import QuantumCircuit
+from qiskit import QuantumJob
+from qiskit import _openquantumcompiler as openquantumcompiler
+from qiskit import JobProcessor
+from qiskit._jobprocessor import run_backend
+from .common import QiskitTestCase
+from ._random_circuit_generator import RandomCircuitGenerator
+
+try:
+    pq_simulator = projectq_simulator.ProjectQSimulator()
+except Exception as err:
+    _skip_class = True
+else:
+    _skip_class = False
+
+@unittest.skipIf(_skip_class, 'Project Q C++ simulator unavailable')
+class TestProjectQCppSimulator(QiskitTestCase):
+    """
+    Test job_pocessor module.
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        super(TestProjectQCppSimulator, cls).setUpClass()
+        nCircuits = 20
+        minDepth = 1
+        maxDepth = 10
+        minQubits = 1
+        maxQubits = 4
+        randomCircuits = RandomCircuitGenerator(minQubits=minQubits,
+                                                maxQubits=maxQubits,
+                                                minDepth=minDepth,
+                                                maxDepth=maxDepth,
+                                                seed=None)
+        for i in range(nCircuits):
+            basis = list(random.sample(randomCircuits.opSignature.keys(),
+                                       random.randint(2,7)))
+            if 'reset' in basis:
+                basis.remove('reset')
+            randomCircuits.add_circuits(1, basis=basis)
+        cls.rqg = randomCircuits
+        cls.seed = 88
+        cls.qasmFileName = os.path.join(qiskit.__path__[0],
+                                        '../test/python/qasm/example.qasm')
```

Comment:
Can you change it to make it a bit more robust for path handling with something similar to:

```
cls.qasmFileName = cls._get_resource_path('qasm/example.qasm')
```

Reply:
That looks much better.